examples/vision/md/randaugment.md

```diff
@@ -0,0 +1,421 @@
+
+
+```python
+rand_aug_model = get_training_model()
+rand_aug_model.load_weights("initial_weights.h5")
+rand_aug_model.compile(
+    loss="sparse_categorical_crossentropy", optimizer="adam", metrics=["accuracy"]
+)
+rand_aug_model.fit(train_ds_rand, validation_data=test_ds, epochs=EPOCHS)
+_, test_acc = rand_aug_model.evaluate(test_ds)
+print("Test accuracy: {:.2f}%".format(test_acc * 100))
+```
+
+<div class="k-default-codeblock">
+```
+391/391 [==============================] - 99s 203ms/step - loss: 2.0931 - accuracy: 0.2734 - val_loss: 1.7154 - val_accuracy: 0.4402
+79/79 [==============================] - 2s 21ms/step - loss: 1.7154 - accuracy: 0.4402
+Test accuracy: 44.02%
+
+```
+</div>
+# 2. Train model with `simple_aug`
+
+
+```python
+simple_aug_model = get_training_model()
+simple_aug_model.load_weights("initial_weights.h5")
+simple_aug_model.compile(
+    loss="sparse_categorical_crossentropy", optimizer="adam", metrics=["accuracy"]
+)
+simple_aug_model.fit(train_ds_simple, validation_data=test_ds, epochs=EPOCHS)
+_, test_acc = simple_aug_model.evaluate(test_ds)
+print("Test accuracy: {:.2f}%".format(test_acc * 100))
+```
+
+<div class="k-default-codeblock">
+```
+391/391 [==============================] - 31s 69ms/step - loss: 1.7836 - accuracy: 0.3788 - val_loss: 1.3316 - val_accuracy: 0.5205
+79/79 [==============================] - 2s 22ms/step - loss: 1.3316 - accuracy: 0.5205
+Test accuracy: 52.05%
+
+```
+</div>
+---
+## Load the CIFAR-10-C dataset and evaluate performance
+
+
+```python
+# Load and prepare the CIFAR-10-C dataset
+# (If it's not already downloaded, it takes ~10 minutes of time to download)
+cifar_10_c = tfds.load("cifar10_corrupted/saturate_5", split="test", as_supervised=True)
+cifar_10_c = cifar_10_c.batch(BATCH_SIZE).map(
+    lambda x, y: (tf.image.resize(x, (IMAGE_SIZE, IMAGE_SIZE)), y),
+    num_parallel_calls=AUTO,
+)
+
+# Evaluate `rand_aug_model`
+_, test_acc = rand_aug_model.evaluate(cifar_10_c, verbose=0)
+print(
+    "Accuracy with RandAugment on CIFAR-10-C (saturate_5): {:.2f}%".format(
+        test_acc * 100
+    )
+)
+
+# Evaluate `simple_aug_model`
+_, test_acc = simple_aug_model.evaluate(cifar_10_c, verbose=0)
+print(
+    "Accuracy with simple_aug on CIFAR-10-C (saturate_5): {:.2f}%".format(
+        test_acc * 100
+    )
+)
+```
+
+<div class="k-default-codeblock">
+```
+Accuracy with RandAugment on CIFAR-10-C (saturate_5): 34.14%
+Accuracy with simple_aug on CIFAR-10-C (saturate_5): 46.90%
+
+```
+</div>
+For the purpose of this example, we trained the models for only a single epoch. In my
+experiments, I found that with RandAugment the model performs way better (76.64%) than
+the model trained with `simple_aug` (64.80%) on the CIFAR-10-C dataset. Additionally, I
+found that RandAugment helped stabilize the training. You can my experimentation notebook
+[here](https://nbviewer.jupyter.org/github/sayakpaul/Keras-Examples-RandAugment/blob/main/RandAugment.ipynb).
```

Comment:
Suggestion: 

- refactor: more "we" and/or "you" and less "I" without changing the meaning 🤷 
- missing word/refactor: "You can my experimentation" -> "You can explore this
[notebook](...)"

```suggestion
For the purpose of this example, we trained the models for only a single epoch. On the
CIFAR-10-C dataset, the model with RandAugment can perform better with a higher accuracy
(for example, 76.64% in one experiment) compared with the model trained with `simple_aug` 
(e.g., 64.80%). RandAugment can also help stabilize the training. You can explore this
[notebook](https://nbviewer.jupyter.org/github/sayakpaul/Keras-Examples-RandAugment/blob/main/RandAugment.ipynb) to check some of the results.
```